app/editor/src/features/admin/media-types/MediaType.tsx

```diff
@@ -0,0 +1,125 @@
+import { FieldSize, IconButton, LabelPosition } from 'components/form';
+import { FormikCheckbox, FormikForm, FormikText } from 'components/formik';
+import { Modal } from 'components/modal';
+import { useModal } from 'hooks';
+import { IMediaTypeModel } from 'hooks/api-editor';
+import React from 'react';
+import { useLocation, useNavigate, useParams } from 'react-router-dom';
+import { toast } from 'react-toastify';
+import { useMediaTypes } from 'store/hooks/admin';
+import { Button, ButtonVariant } from 'tno-core';
+import { Row } from 'tno-core/dist/components/flex';
+
+import { defaultMediaType, MEDIA_LIST_VIEW_ROUTE } from './constants';
+import * as styled from './styled';
+
+/** The page used to view and edit media types in the administrative section. */
+export const MediaType: React.FC = () => {
+  const [, api] = useMediaTypes();
+  const { state } = useLocation();
+  const { id } = useParams();
+  const navigate = useNavigate();
+  const mediaTypeId = Number(id);
+  const [mediaType, setMediaType] = React.useState<IMediaTypeModel>(
+    (state as any)?.mediaType ?? defaultMediaType,
+  );
+
+  const { toggle, isShowing } = useModal();
```

Comment:
For organization and standards.  Keep all your `use*()` statements at the top of the component/hook.   Then a space.
Then all your `useState()` statements grouped together.
Then a space.
Then your custom variables needs from the above.
Then a space.
Then all your `useEffects()` statements grouped together.
Then a space.
Then all your functions and handler statements grouped together.